src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,60 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private final int[] array;
+    private final int start;
+    private final int end;
 
     public MergeSortAction(int[] array) {
+        this(array, 0, array.length);
+    }
+
+    private MergeSortAction(int[] array, int start, int end) {
+        this.array = array;
+        this.start = start;
+        this.end = end;
+    }
+
+    @Override
+    protected void compute() {
+        if (end - start < 2) {
+            return;
+        }
+
+        int middle = start + (end - start) / 2;
+
+        MergeSortAction left = new MergeSortAction(array, start, middle);
+        MergeSortAction right = new MergeSortAction(array, middle, end);
+
+        invokeAll(left, right);
+
+        merge(start, middle, end);
+    }
+
+    public void merge(int start, int middle, int end) {
```

Comment:
Consider making the `merge` method private since it is only used within the `MergeSortAction` class. This would encapsulate the functionality better and prevent external access.